File: dotnet/console/Console/Scenarios/CreatePDFDocumentWithCadesSignatureScenario.cs
using Lacuna.Signer.Api;
using Lacuna.Signer.Api.Documents;
using Lacuna.Signer.Api.FlowActions;
using Lacuna.Signer.Api.Users;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    public class CreatePDFDocumentWithCadesSignatureScenario : Scenario
    {
        /**
         * This scenario shows step-by-step the submission of a document
         * to the signer instance where this document it's a PDF and needs to be
         * signed with Cades.
         */
        public override async Task RunAsync()
        {
            // 1. The file's bytes must be read by the application and uploaded
            var filePath = "sample.pdf";
            var fileName = Path.GetFileName(filePath);
            var file = File.ReadAllBytes(filePath);
            var uploadModel = await signerClient.UploadFileAsync(fileName, file, "application/pdf");

            // 2. Define the name of the document which will be visible in the application
            var fileUploadModel = new FileUploadModel(uploadModel) { DisplayName = "PDF Cades Sample" };

            // 3. For each participant on the flow, create one instance of ParticipantUserModel.
            var participantUser = new ParticipantUserModel()
            {
                Name = "Jack Bauer",
                Email = "jack.bauer@mailinator.com",
                Identifier = "75502846369"
            };

            // 4. Create a FlowActionCreateModel instance for each action (signature or approval) in the flow.
            //    This object is responsible for defining the personal data of the participant and the type of 
            //    action that he will peform on the flow.
            var flowActionCreateModel = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = participantUser
            };

            // 5. Send the document create request specifying that it requires CAdES signatures, since PAdES is
            //    the default for PDF files.
            var documentRequest = new CreateDocumentRequest()
            {
                Files = new List<FileUploadModel>() { fileUploadModel },
                FlowActions = new List<FlowActionCreateModel>() { flowActionCreateModel },
                ForceCadesSignature = true
            };
            var result = (await signerClient.CreateDocumentAsync(documentRequest)).First();

            System.Console.WriteLine($"Document {result.DocumentId} created");
        }
    }
}

File: nodejs/scenarios/createPDFDocumentWithCadesSignatureScenario.ts
import {
  DocumentsApi,
  FileUploadModel,
  UploadApi,
  UsersParticipantUserModel,
  FlowActionsFlowActionCreateModel,
  FlowActionType,
  DocumentsCreateDocumentRequest,
  getBase64,
} from "signer-node-client";
import { config } from "./scenario";

const uploadApi = new UploadApi(config);
const documentsApi = new DocumentsApi(config);
const filepath = "../samples/sample.pdf";
const filename = "sample.pdf";

// 1. The file's bytes must be read by the application and uploaded
uploadApi.apiUploadsBytesPost({ bytes: getBase64(filepath) }).then((res) => {
  // 2. Define the name of the document which will be visible in the application
  const uploadModel: FileUploadModel = {
    id: res.data.id,
    name: filename,
    contentType: "application/pdf",
    displayName: "Cades Signature Sample",
  };
  // 3. For each participant on the flow, create one instance of ParticipantUserModel
  const participant: UsersParticipantUserModel = {
    name: "Jack Bauer",
    email: "jack.bauer@mailinator.com",
    identifier: "75502846369",
  };
  // 4. Create a FlowActionCreateModel instance for each action (signature or approval) in the flow.
  //    This object is responsible for defining the personal data of the participant and the type of
  //    action that he will perform on the flow
  const flowAction: FlowActionsFlowActionCreateModel = {
    type: FlowActionType.Signer,
    user: participant,
  };

  // 5. Send the document create request
  const documentRequest: DocumentsCreateDocumentRequest = {
    files: [uploadModel],
    flowActions: [flowAction],
    // 6. This time we'll add the forceCadesSignature parameter
    forceCadesSignature: true,
  };
  documentsApi.apiDocumentsPost(documentRequest).then((res) => {
    console.log("Document ", res.data[0].documentId, " Created");
  });
});
